x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -568,6 +568,73 @@ public void maybeFetchRegion(
         }
     }
 
+    /**
+     * Fetch and write in cache a range within a blob region if there is at least a free page in the cache to do so.
+     * <p>
+     * This method returns as soon as the download tasks are instantiated, but the tasks themselves
+     * are run on the bulk executor.
+     * <p>
+     * If an exception is thrown from the writer then the cache entry being downloaded is freed
+     * and unlinked
+     *
+     * @param cacheKey      the key to fetch data for
+     * @param region        the region of the blob
+     * @param range         the range in the region to fetch
```

Comment:
I think this should be "the range in the blob to fetch"? At least the wording here made me think it was relative to the region start.

Reply:
Right, I pushed 970394895889a908029084156569704e4b207dc5.

I must admit I took a shortcut here by allowing to pass a range of the blob that cannot be included within the `region`, letting the method implementation to remap the range to the region and call the listener on empty range.

I think we do this in other places, not great, but it simplifies the calling methods.